server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -745,6 +748,108 @@ public DocTableInfo dropColumns(List<DropColumn> columns) {
         );
     }
 
+    private void validateRenameColumn(Reference refToRename, ColumnIdent newName) {
+        var oldName = refToRename.column();
+        var reference = getReference(oldName);
+        if (reference == null) {
+            reference = indexColumn(oldName);
+        }
+        if (!refToRename.equals(reference)) {
+            throw new ColumnUnknownException(oldName, ident);
+        }
+        if (getReference(newName) != null || indexColumn(newName) != null) {
+            throw new IllegalArgumentException("Cannot rename column to a name that is in use");
+        }
+    }
+
+    public DocTableInfo renameColumn(Reference refToRename, ColumnIdent newName) {
+        validateRenameColumn(refToRename, newName);
+        final ColumnIdent oldName = refToRename.column();
+
+        Predicate<ColumnIdent> toBeRenamed = c -> c.equals(oldName) || c.isChildOf(oldName);
+
+        // Renaming columns are done in 2 steps:
+        //      1) rename SimpleReferences' own ColumnIdents
+        //      2) rename dependencies such as GeneratedReferences, IndexReferences, Check Constraints, etc.
+        // where 1) is used to perform 2).
+
+        Map<ColumnIdent, Reference> oldNameToRenamedRefs = new HashMap<>();
+        for (var ref : references.values()) {
+            var r = toBeRenamed.test(ref.column()) ? ref.withReferenceIdent(
+                new ReferenceIdent(ref.ident().tableIdent(), ref.column().replacePath(newName))) : ref;
+            oldNameToRenamedRefs.put(
+                ref.column(), // old name
+                r); // renamed reference
+        }
+
+        Function<Reference, Reference> renameGeneratedRefs = ref -> {
+            if (ref instanceof GeneratedReference genRef) {
+                return new GeneratedReference(
+                    genRef.reference(), // already renamed in step 1)
+                    RefReplacer.replaceRefs(genRef.generatedExpression(), r -> oldNameToRenamedRefs.getOrDefault(r.column(), r))
+                );
+            }
+            return ref;
+        };
+
+        Function<IndexReference, IndexReference> renameIndexRefs = idxRef -> {
+            var dependenciesRenamed = idxRef.updateColumns(idxRef.columns().stream()
+                    .map(r -> oldNameToRenamedRefs.getOrDefault(r.column(), r)).toList());
+            if (toBeRenamed.test(idxRef.column())) {
+                return (IndexReference) dependenciesRenamed.withReferenceIdent(
+                    new ReferenceIdent(idxRef.ident().tableIdent(), idxRef.column().replacePath(newName)));
+            }
+            return dependenciesRenamed;
```

Comment:
`dependenciesRenamed` sounds like a collection due to the use of plural. Could use either `IndexReference dependenciesRenamed` or just `var updatedRef`